src/libs/shim-services/src/lib.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! shim-services is a common library for different components of Kata Containers
+//! to make function call through services inside the shim.
+//!
+//! Shim management:
+//! Currently, inside the shim, there is a shim management server running as the shim
+//! starts, working as a RESTful server. To make function call in shim from another
+//! binary, using the utilities provided in this library is one of the methods.
+//!
+//! You may construct clients by construct a MgmtClient and let is make specific
+//! HTTP request to the server. The server inside shim will multiplex the request
+//! to its corresponding handler and run certain methods.
+
+use std::path::Path;
+
+pub mod shim_mgmt;
+
+pub const KATA_PATH: &str = "/run/kata";
+pub const SHIM_MGMT_SOCK_NAME: &str = "shim-monitor.sock";
+
+// return sandbox's storage path
+pub fn sb_storage_path() -> String {
+    String::from(KATA_PATH)
+}
+
+// returns the address of the unix domain socket(UDS) for communication with shim
+// management service using http
+// normally returns "unix:///run/kata/{sid}/shim_monitor.sock"
+pub fn mgmt_socket_addr(sid: String) -> String {
```

Comment:
Should the `sid` value be verified in this function? what if `sid = "../../tmp/malicous-hacker-directory/"` ? it can use https://github.com/kata-containers/kata-containers/tree/main/src/libs/safe-path

Reply:
True, I think we also need to check some of the possible places where we can appropriately use this.

Reply:
No, this socket address is used when some endpoint want to connect it. If the address is not valid, the connection will surely fail.

Reply:
Yes, but I think you need to check if that is valid and return some error information for users

Reply:
Still I dont think that is necessary. This functions is equivalent to formatting a `sid` to a `uds`. It is the caller's responsibility to check the validity of socket address.

Reply:
fair enough